### PR TITLE
Prepare for release

### PR DESCRIPTION
**Pull request by msluszniak**
Mainly fixes in documentation. I also added changelog.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -30,9 +30,9 @@ defmodule Scholar.MixProject do
   defp deps do
     [
       {:ex_doc, "~> 0.29", only: :docs},
-      {:nx, github: "elixir-nx/nx", sparse: "nx", override: true},
+      {:nx, "~> 0.6", override: true},
       {:nimble_options, "~> 0.5.2 or ~> 1.0"},
-      {:exla, github: "elixir-nx/exla", sparse: "exla", only: [:test, :dev]},
+      {:exla, github: "elixir-nx/exla", sparse: "exla", only: [:test, :dev], override: true},
```
Comment (msluszniak) [inline mix.exs:35]: I don't if this line should stay like this

Comment (josevalim) [reply mix.exs:35]: We should depend on it as regular Hex package! Any reason not to?

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Changelog
+
+## v0.2.1-dev
+
+## v0.2.0 (2023-08-16)
+
+This version requires Elixir v1.14.5+.
```
Comment (josevalim) [inline CHANGELOG.md:7]:
```suggestion
This version requires Elixir v1.14+.
```

**Review by josevalim — APPROVED** [review]

File: mix.exs (hunk 1 above)
Comment (msluszniak) [reply mix.exs]: Ok, I needed to update `xla` in `mix.lock`. A also bumped ex_doc to 0.30.

**Pull request merged**